lib/cppcheck.cpp

```diff
@@ -350,6 +364,9 @@ unsigned int CppCheck::check(const std::string &path)
         for (const std::string &i: mSettings.includePaths)
             flags += "-I" + i + " ";
 
+        flags += getDefinesFlags(mSettings.userDefines);
```

Comment:
hmm .. an alternative could be to reuse `split`.

    for (const std::string &d: split(mSettings.userDefines, ";"))
        flags += "-D" + d + " ";

Reply:
please reuse `split`.. if you do it here like my code or rewrite your `getDefinesFlags` that is up to you :-)